bazel/repositories.bzl

```diff
@@ -91,19 +91,28 @@ cc_library(
     visibility = ["//visibility:public"],
 )
 
+# There is some moderate evil here. buildconf needs to be invoked in its source
+# location, where it also generates temp files (no option to place in TMPDIR).
+# Also, configure does a cd relative dance that doesn't play nice with the
+# symlink execroot in the Bazel build. So, copy everything prior to building.
+ARES_CONFIG_SH_CMDS = [
+    "export TMPDIR=$$(realpath $(@D))",
+    "rm -rf $(@D)/cares_build",
+    "mkdir -p $(@D)/cares_build",
+    "rsync -aRLv $(SRCS) $(@D)/cares_build",
+    "pushd $(@D)/cares_build/external/cares_git",
+    "./buildconf",
+    "./configure",
+    "popd",
+    "cp $(@D)/cares_build/external/cares_git/ares_config.h $@",
+]
+
 genrule(
     name = "config",
     # This list must not contain ares_config.h.
-    srcs = glob(["**/*.m4"]),
```

Comment:
We need more files now that we're not illegally stealing them from an internal directory :) It doesn't have to be that tight a bound, this glob only hits everything in the c-ares source tree.